optuna/integration/botorch.py

```diff
@@ -8,6 +8,7 @@
 import warnings
 
 import numpy
+from pkg_resources import parse_version
```

Comment:
Just a curiosity, but some other modules use `packaging.version` instead of `pkg_resouses.parse_version`. I

https://github.com/optuna/optuna/blob/906f6a6f2bf8d0e0700ece482c1b666c636076c7/optuna/visualization/matplotlib/_matplotlib_imports.py#L1
https://github.com/optuna/optuna/blob/906f6a6f2bf8d0e0700ece482c1b666c636076c7/optuna/visualization/_plotly_imports.py#L1

Could you tell me why you chose `pkg_resource` in this case?

Reply:
I have no particular reason. I think both are OK.

Reply:
[pkg_resource](https://github.com/pypa/pkg_resources/tree/main) package is no more active while [packaging](https://github.com/pypa/packaging) is active. I'm not sure, but it would be better to use `packaging`.

Reply:
`pkg_resources` seems to be developing in `setuptools`. https://github.com/pypa/setuptools/tree/main/pkg_resources
By the way, unifying with `packaging` could lower maintenance costs later on. I'll update this PR.